preferencesfx/src/main/java/com/dlsc/preferencesfx/util/PreferencesBasedStorageHandler.java

```diff
@@ -0,0 +1,297 @@
+    preferences.putDouble(WINDOW_POS_X, windowPosX);
+  }
+
+  /**
+   * Searches for the horizontal window position.
+   *
+   * @return the double value of the horizontal window position
+   */
+  public double loadWindowPosX() {
+    return preferences.getDouble(WINDOW_POS_X, DEFAULT_PREFERENCES_POS_X);
+  }
+
+  /**
+   * Stores the position of the PreferencesFxDialog in vertical orientation.
+   *
+   * @param windowPosY the double value of the window position in vertical orientation
+   */
+  public void saveWindowPosY(double windowPosY) {
+    preferences.putDouble(WINDOW_POS_Y, windowPosY);
+  }
+
+  /**
+   * Searches for the vertical window position.
+   *
+   * @return the double value of the vertical window position
+   */
+  public double loadWindowPosY() {
+    return preferences.getDouble(WINDOW_POS_Y, DEFAULT_PREFERENCES_POS_Y);
+  }
+
+  /**
+   * Serializes a given Object and saves it to the preferences using the given key.
+   *
+   * @param breadcrumb the key which is used to save the serialized Object
+   * @param object     the Object which will be saved
+   */
+  public void saveObject(String breadcrumb, Object object) {
+    preferences.put(hash(breadcrumb), serialize(object));
+  }
+
+  /**
+   * Searches in the preferences after a serialized Object using the given key,
+   * deserializes and returns it. Returns a default Object if nothing is found.
+   *
+   * @param breadcrumb    the key which is used to search the serialized Object
+   * @param defaultObject the Object which will be returned if nothing is found
+   * @return the deserialized Object or the default Object if nothing is found
+   */
+  public Object loadObject(String breadcrumb, Object defaultObject) {
+    String json = getSerializedPreferencesValue(breadcrumb, serialize(defaultObject));
+    return deserialize(json, Object.class);
+  }
+
+  /**
+   * Searches in the preferences after a serialized ArrayList using the given key,
+   * deserializes and returns it as ObservableArrayList.
+   * When an ObservableList is deserialzed, Gson returns an ArrayList
+   * and needs to be wrapped into an ObservableArrayList. This is only needed for loading.
+   *
+   * @param breadcrumb            the key which is used to search the serialized ArrayList
+   * @param defaultObservableList the default ObservableList
+   *                              which will be returned if nothing is found
+   * @return the deserialized ObservableList or the default ObservableList if nothing is found
+   */
+  public ObservableList loadObservableList(
+      String breadcrumb,
+      ObservableList defaultObservableList
+  ) {
+    String json = getSerializedPreferencesValue(breadcrumb, serialize(defaultObservableList));
+    return FXCollections.observableArrayList(deserialize(json, ArrayList.class));
+  }
+
+  private String getSerializedPreferencesValue(String breadcrumb, String serializedDefault) {
+    String json = preferences.get(hash(breadcrumb), serializedDefault);
```

Comment:
maybe, now that the serialized output could be literally anything (even `xml` or what not), maybe it would make sense to refactor `json` as `serialized` here, to not give the impression that it must be some kind of JSON?

Reply:
fixed